### ISSUE TITLE
Dependecy issues of NetCDF and numpy with mixture of poetry and nox

### ISSUE THREAD

**Issue by cchwala** [OpenSenseAction/poligrain#4]
As can be seen by the failing test from #2 [here](https://github.com/OpenSenseAction/poligrain/actions/runs/7535405172/job/20511361456?pr=2), there is a problem which probably stems from incompatible binaries of NetCDF and numpy.

If I go into to poetry env via `poetry run python`, I can successfully  do the following
```python
import xarray as xr
xr.open_dataset('tests/test_data/openMRG_CML_180minutes.nc')
```
which is the same thing that is done where the test on CI fails. This also fails when test are run via `nox -s tests` locally.

Relevant package versions are:

```
poetry run pip freeze
```

```
matplotlib==3.8.2
netCDF4==1.6.5
numpy==1.26.3
```

using this in `pyproject.toml`
```
netcdf4 = "^1.6.5"
xarray = "^2023.12.0"
matplotlib = "^3.8.2"
```

The `pip freeze` output that I added for `nox -s tests` shows the same
```
matplotlib==3.8.2
netCDF4==1.6.5
numpy==1.26.3
```

But the test does this
```python
.nox/tests/lib/python3.10/site-packages/netCDF4/__init__.py:3: in <module>
    from ._netCDF4 import *
        __builtins__ = <builtins>
        __cached__ = '/Users/chwala-c/code/poligrain/.nox/tests/lib/python3.10/site-packages/netCDF4/__pycache__/__init__.cpython-310.pyc'
        __doc__    = None
        __file__   = '/Users/chwala-c/code/poligrain/.nox/tests/lib/python3.10/site-packages/netCDF4/__init__.py'
        __loader__ = <_frozen_importlib_external.SourceFileLoader object at 0x121303760>
        __name__   = 'netCDF4'
        __package__ = 'netCDF4'
        __path__   = ['/Users/chwala-c/code/poligrain/.nox/tests/lib/python3.10/site-packages/netCDF4']
        __spec__   = ModuleSpec(name='netCDF4', loader=<_frozen_importlib_external.SourceFileLoader object at 0x121303760>, origin='/Users/...t__.py', submodule_search_locations=['/Users/chwala-c/code/poligrain/.nox/tests/lib/python3.10/site-packages/netCDF4'])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 16 from C header, got 96 from PyObject


src/netCDF4/_netCDF4.pyx:1: RuntimeWarning
```

**Comment by cchwala** [comment]
The problem seems to stem from an interplay of numpy and netCDF4. See https://github.com/pydata/xarray/issues/7259

**Comment by cchwala** [comment]
Should be fixed with the hack in 41e238ab27e7dee577328d1c138ad8a81df410a6

**Comment by cchwala** [comment]
I dropped Python3.12 support in #2 

I close this issue, but at some point the hack with the import of netCDF4 should be reverted.
